src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -964,7 +964,7 @@ CollectionExpressionTypeKind.Array or CollectionExpressionTypeKind.Span or Colle
 
                 var implicitReceiver = new BoundObjectOrCollectionValuePlaceholder(syntax, isNewInstance: true, _targetType) { WasCompilerGenerated = true };
 
-                var collectionCreation = bindCollectionConstructorConstruction(in this, syntax, constructor);
+                var collectionCreation = bindCollectionConstructorConstruction(in this, _node.WithElement?.Syntax ?? _node.Syntax, constructor);
```

Comment:
this ensures we give the error on the `with(...)` element if present when trying to bind the collection constructor.